File: HackerRank/Practice/Algorithms/Warmup/SimpleArraySum/SimpleArraySum.java
/*
 * Solution to the "SimpleArraySum" challenge from HackerRank:
 * https://www.hackerrank.com/challenges/simple-array-sum/problem
 * Created on: 17.03.2020
 * Last modified: 17.03.2020
 * Author: Adam Gralinski (adam@gralin.ski)
 * License: MIT
 */

import java.io.*;
import java.util.*;

public class SimpleArraySum
{
  static int simpleArraySum(int[] array) {
    int sum = 0;
    for (int i = 0; i < array.length; ++i) {
      sum += array[i];
    }

    return sum;
  }

  private static final Scanner scanner = new Scanner(System.in);

  public static void main(String[] args) throws IOException {
    int arrSize = Integer.parseInt(scanner.nextLine().trim());
    int[] array = new int[arrSize];

    String[] strArray = scanner.nextLine().split(" ");
    for (int i = 0; i < arrSize; ++i)
    {
      array[i] = Integer.parseInt(strArray[i].trim());
    }

    int sum = simpleArraySum(array);
    System.out.println(String.valueOf(sum));
  }
}

File: HackerRank/Practice/Algorithms/Warmup/SimpleArraySum/cpp/SimpleArraySum.cpp
/**
 * Solution to the "SimpleArraySum" challenge from HackerRank:
 * https://www.hackerrank.com/challenges/simple-array-sum/problem
 * Created on: 18.03.2020
 * Last modified: 18.03.2020
 * Author: Adam Gralinski (adam@gralin.ski)
 * License: MIT
 */

#include <iostream>
#include <vector>

int main(int, char**)
{
  std::size_t N;
  std::cin >> N;
  std::vector<unsigned long long> v(N);
  unsigned long long sum = 0ULL;
  for (std::size_t n = 0; n < N; ++n)
  {
    std::cin >> v[n];
    sum += v[n];
  }
  std::cout << sum << "\n";
  return 0;
}
